src/shaka_scheme/system/gc/GC.hpp

```diff
@@ -0,0 +1,38 @@
+//
+// Created by Brandon Wong on 3/26/18.
+//
+
+#ifndef SHAKA_SCHEME_GC_HPP
+#define SHAKA_SCHEME_GC_HPP
+
+#include "shaka_scheme/system/gc/GCList.hpp"
+#include "shaka_scheme/system/gc/GCData.hpp"
+#include "shaka_scheme/system/base/Data.hpp"
+
+namespace shaka {
+
+    namespace gc {
+
+    /**
+         * @brief Implements the factory method of the GC, which allows the
+         * construction of GCData and automatically adds those GCData to the
+         * GCList
+         */
+        class GC {
+        public:
+
+            GC();
+            ~GC();
+            GC(const GC& other) = delete;
+            GC(GC&& other);
+
+            GCData *create_data(const shaka::Data& data);
+            int get_size();
```

Comment:
This should be a `const` member function so that we can still call it if we are referring to `GC` as a `const` reference or variable within some sort of scope.

Reply:
@btwooton

Reply:
Just declare it as int get_size const; Yiu can declare a method as const if it does not change any of the member variables of the object when it is called. It’s a best practice to do so when you can. Sorry I never explained this to you until now.

Reply:
Are you saying we are redefining the method, but it still does the same thing in getting the size of the list? `int get_size const()`